pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -212,20 +242,215 @@ func (c *MCDefaultRouteController) processNextWorkItem() bool {
 	}
 	defer c.queue.Done(obj)
 
-	if k, ok := obj.(string); !ok {
+	key, ok := obj.(string)
+	if !ok {
 		c.queue.Forget(obj)
 		klog.InfoS("Expected string in work queue but got", "object", obj)
 		return true
-	} else if err := c.syncMCFlows(); err == nil {
-		c.queue.Forget(k)
+	}
+	if c.wireGuardConfig != nil {
+		wireGuardWasInitialized := c.wireGuardInitialized
+		if err := c.reconcileWireGuard(key); err != nil {
+			klog.ErrorS(err, "Failed to reconcile Multicluster WireGuard")
+			c.queue.AddRateLimited(key)
+			return true
+		}
+
+		// If the WireGuard was not initialized before and it's initialized after reconcileWireGuard, it's means
+		// that the Gateway has been updated and a new Gateway event will be triggered. We could just skip syncMCFlows.
+		if !wireGuardWasInitialized && c.wireGuardInitialized {
+			klog.InfoS("The Gateway has been updated")
```

Comment:
Probably `klog.V(2).InfoS("The Gateway has been updated, do nothing")`?

Reply:
Changed.